packages/std/src/coin.rs

```diff
@@ -55,6 +55,31 @@ impl fmt::Display for Coin {
     }
 }
 
+/// Equivalent to Coin, but with a reference to the denom
+#[derive(Serialize, Deserialize, Debug, Clone, Default, PartialEq, Eq, JsonSchema)]
+pub struct CoinRef<'a> {
+    pub denom: &'a str,
+    pub amount: Uint128,
+}
+
+impl From<CoinRef<'_>> for Coin {
+    fn from(value: CoinRef) -> Self {
+        Coin {
+            denom: value.denom.to_owned(),
+            amount: value.amount,
+        }
+    }
+}
+
+impl<'a> From<&'a Coin> for CoinRef<'a> {
+    fn from(value: &'a Coin) -> Self {
+        CoinRef {
+            denom: &value.denom,
+            amount: value.amount,
+        }
+    }
+}
```

Comment:
For some reason I am not convinced this is the best way to achieve what we want. Shouldn't we have a `trait CoinLike` that is implemented by all 3 Coin, &Coin and CoinRef instead?

Reply:
Yes, that's also an option. It would do effectively the same thing: Provide you access to the `amount` and `&denom`. Maybe as separate functions instead of one that returns both in a `CoinRef`.
The advantage would be that `impl CoinLike` is a little shorter than `impl Into<CoinRef<'a>>` and avoids the cumbersome `.into()` call. I like it.
I still think it's good to have a `From` impl anyways, when the conversion is so trivial, as it allows for better composability with anything that works with `From<T>`.